src/ACTSMergeHitCollections.cxx

```diff
@@ -0,0 +1,37 @@
+#include "ACTSMergeHitCollections.hxx"
+
+// edm4hep
+#include <edm4hep/TrackerHitPlane.h>
+
+
+DECLARE_COMPONENT(ACTSMergeHitCollections)
+
+
+ACTSMergeHitCollections::ACTSMergeHitCollections(const std::string& name, ISvcLocator* svcLoc) : MultiTransformer(name, svcLoc, {
+		KeyValue("InputCollection1", "Collection1"),
+		KeyValue("InputCollection2", "Collection2"),
+		KeyValue("InputCollection3", "Collection3"),
+		KeyValue("InputCollection4", "Collection4"),
+		KeyValue("InputCollection5", "Collection5"),
+		KeyValue("InputCollection6", "Collection6") },
+	      { KeyValue("OutputCollection", "MergedCollection") }) {}
+
+std::tuple<edm4hep::TrackerHitPlaneCollection> ACTSMergeHitCollections::operator()(
+		const DataWrapper<edm4hep::TrackerHitPlaneCollection>& col1,
+		const DataWrapper<edm4hep::TrackerHitPlaneCollection>& col2,
+		const DataWrapper<edm4hep::TrackerHitPlaneCollection>& col3,
+		const DataWrapper<edm4hep::TrackerHitPlaneCollection>& col4,
+		const DataWrapper<edm4hep::TrackerHitPlaneCollection>& col5,
+		const DataWrapper<edm4hep::TrackerHitPlaneCollection>& col6) const{
+	// Initialize collection
+	edm4hep::TrackerHitPlaneCollection mergedCollection;
+
+	// Loop over all items in all collections and add each to a new collection
+	for (const auto& col : {col1.getData(), col2.getData(), col3.getData(), col4.getData(), col5.getData(), col6.getData()}) {
+		for (const auto& item : *col) {
+			(void)mergedCollection->create(item.getCellID(), item.getType(), item.getQuality(), item.getTime(), item.getEDep(), item.getEDepError(), item.getU(), item.getV(), item.getDu(), item.getDv(), item.getPosition(), item.getCovMatrix());
+		}
+	}
```

Comment:
```suggestion
	edm4hep::TrackerHitPlaneCollection mergedCollection;
        mergedCollection.setSubsetCollection();
        
	// Loop over all items in all collections and add each to a new collection
	for (const auto& col : {col1.getData(), col2.getData(), col3.getData(), col4.getData(), col5.getData(), col6.getData()}) {
		for (const auto& item : *col) {
			mergedCollection->push_back(item);
		}
	}
```
This way you don't duplicate the information, but rather make the returned collection be a collection that simply points to the original objects. `subsetCollection` is not the best name in this case, since it is technically a superset collection here.

This is also one of the things where the `k4FWCore` implemenation of the Functional algorithms can deal with an arbitrary number of input collections, see here for a `Consumer` example 
[c++side](https://github.com/key4hep/k4FWCore/blob/9b6949492a52fd8db97ba043499359d504d78485/test/k4FWCoreTest/src/components/ExampleFunctionalConsumerRuntimeCollectionsMultiple.cpp#L42-L80)
[python side](https://github.com/key4hep/k4FWCore/blob/9b6949492a52fd8db97ba043499359d504d78485/test/k4FWCoreTest/options/ExampleFunctionalConsumerRuntimeCollectionsMultiple.py#L63-L75)

Reply:
The structure of this module is getting more and more complicated.
I'm wondering if it's worth creating a different git repo for the guadi implementation. Each module can define its own dependencies; it does not make sense for the guadi implementation to depend on Marlin.
It there're pieces of code shared between the two modules we can move them in third common repo.

Reply:
In principle we have https://github.com/key4hep/k4ActsTracking where we plan to have things for key4hep

Reply:
For the k4FWCore functionals, I initially started with those, but the muon collider container's version of Key4HEP doesn't have those yet. So for compatibility I stuck with the Gaudi ones. Shouldn't be too hard to switch when the container updates. 

As for the subsetCollection -- that sounds great! Do I have to do anything different in handling the reading of them?

Reply:
> As for the subsetCollection -- that sounds great! Do I have to do anything different in handling the reading of them?

No, they are handled completely transparently on the I/O side. I.e. for reading you can treat it as a "normal" collection *iff* all the elements this subset collection refers to are also written. I.e. if you start dropping some of the original hit collections when writing, they will obviously also not be available in the subset collection when you read them back.

> For the k4FWCore functionals, I initially started with those, but the muon collider container's version of Key4HEP doesn't have those yet. So for compatibility I stuck with the Gaudi ones. Shouldn't be too hard to switch when the container updates.

That makes sense. And I agree that shouldn't be a very complicated change.